### PR TITLE
Added session argument to Navigator.hal factory method.

### PR DESCRIPTION
**Pull request by tspng**
That's my first pull request, hope it works.

I also updated the authentication part of the documentation, hope that's ok like that (English is not my mother tongue).

### CODE REVIEW

**Comment by bubenkoff** [comment]
please add small test for this change

**Comment by tspng** [comment]
Sorry, completely forgot about the tests.
Class TestCreate in test_hal_nav2py would be the correct place, right?

**Comment by bubenkoff** [comment]
i think just a test_session(
a function in that module

On 30 April 2015 at 12:09, haestan notifications@github.com wrote:

> Sorry, completely forgot about the tests.
> Class TestCreate in test_hal_nav2py would be the correct place, right?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/deontologician/restnavigator/pull/38#issuecomment-97727602
> .

## 

Anatoly Bubenkov

**Comment by tspng** [comment]
Right, TestCreate is about the create() method, my bad.

File: restnavigator/halnav.py [hunk 1]
```diff
@@ -167,7 +167,13 @@ class Navigator(object):
     '''
 
     @staticmethod
-    def hal(root, apiname=None, default_curie=None, auth=None, headers=None):
+    def hal(root, 
+            apiname=None, 
+            default_curie=None,
+            auth=None, 
+            headers=None, 
+            session=None
```
Comment (deontologician) [inline restnavigator/halnav.py]: could you add a trailing comma after this last line? if we add more keyword args in the future, we won't have to modify it

**Comment by deontologician** [comment]
This is looking good! I like that you added an example, that's awesome

File: README.rst [hunk 2]
```diff
@@ -262,6 +262,18 @@ Now we can put it all together to create a new post:
     >>> N_post()
     {'content': 'My first post', 'created_at': '2015-06-13T19:38:59+00:00'}
 
+It is also possible to specify a custom requests Session object when creating 
+a new navigator.
+
+For example, if you want to talk to a OAuth2 protected api, simply pass
```
Comment (deontologician) [inline README.rst:268]: Should be "an OAuth2", otherwise the grammar is perfect

**Comment by tspng** [comment]
Ok, fixed the comma and spelling issue.
Added a test case for the session argument, but it's a really stupid test. Just comparing if the passed session argument ends up at _core's session property.

**Comment by deontologician** [comment]
Thanks!

**Pull request merged**